src/cargo/util/toml/targets.rs

```diff
@@ -98,7 +99,7 @@ pub(super) fn to_targets(
         );
         targets.push(Target::custom_build_target(
             &name,
-            package_root.join(custom_build),
+            paths::normalize_path(package_root.join(custom_build).as_path()),
```

Comment:
This is happening during a `to_` operation while all the rest are during a `normalize` operation